### PR TITLE
Renovation: odata/query_adapter es6 refactoring

### PR DESCRIPTION
**Pull request by LazyLahtak**
Part of refactoring odata/* components

(In plans - get rid of odata/mixins)

### CODE REVIEW

File: js/data/odata/query_adapter.js [hunk 1]
```diff
@@ -157,11 +155,11 @@ const createODataQueryAdapter = function(queryOptions) {
 
     const _oDataVersion = queryOptions.version || DEFAULT_PROTOCOL_VERSION;
 
-    const hasSlice = function() {
+    const hasSlice = () => {
```
Comment (churkin) [inline js/data/odata/query_adapter.js]: Use one line function definition for simple lambdas
```javascript
const hasSlice = () => _skip || _take !== undefined;
```

File: js/data/odata/query_adapter.js [hunk 2]
```diff
@@ -232,19 +230,19 @@ const createODataQueryAdapter = function(queryOptions) {
 
         tasks[1 + selectIndex] = tasks[selectIndex];
         tasks[selectIndex] = nextTask;
-    }
+    };
 
     return {
 
-        optimize: function(tasks) {
+        optimize(tasks) {
```
Comment (churkin) [inline js/data/odata/query_adapter.js]:
```javascript
optimize:  tryLiftSelect,
```

File: js/data/odata/query_adapter.js [hunk 3]
```diff
@@ -232,19 +230,19 @@ const createODataQueryAdapter = function(queryOptions) {
...
             tryLiftSelect(tasks);
         },
 
-        exec: function(url) {
-            return odataUtils.sendRequest(_oDataVersion,
+        exec(url) {
+            return sendRequest(_oDataVersion,
                 {
                     url: url,
```
Comment (churkin) [inline js/data/odata/query_adapter.js]: `url,`

File: js/data/odata/query_adapter.js [hunk 4]
```diff
@@ -327,12 +325,12 @@ const createODataQueryAdapter = function(queryOptions) {
             _select = expr;
         },
 
-        count: function() {
+        count() {
```
Comment (churkin) [inline js/data/odata/query_adapter.js]:
```javascript
count: () => _countQuery = true,
```

**Review by churkin — CHANGES_REQUESTED** [review]

File: js/data/odata/query_adapter.js [hunk 5]
```diff
@@ -1,33 +1,31 @@
...
     let protocolVersion;
     let forceLowerCase;
     let fieldTypes;
 
-    const createBinaryOperationFormatter = function(op) {
-        return function(prop, val) {
-            return prop + ' ' + op + ' ' + val;
+    const createBinaryOperationFormatter = (op) => {
+        return (prop, val) => {
+            return `${prop} ${op} ${val}`;
```
Comment (MaximKudriavtsev) [inline js/data/odata/query_adapter.js]: Also `(prop, val) => ${prop} ${op} ${val}`

**Review by MaximKudriavtsev — APPROVED** [review]

**Comment by LazyLahtak** [comment]
Ready to review

**Review by churkin — APPROVED** [review]

**Pull request merged**